perception/lidar_centerpoint_tvm/lib/postprocess/generate_detected_boxes.cpp

```diff
@@ -69,39 +69,39 @@ void generateBoxes3D_worker(
   // heatmap: N = class_size, offset: N = 2, z: N = 1, dim: N = 3, rot: N = 2, vel: N = 2
   for (std::size_t idx = 0; idx < grids_per_thread; idx++) {
     std::size_t grid_idx = thread_idx * grids_per_thread + idx;
+    const auto grid_size = config.grid_size_y_ * config.grid_size_x_;
     const auto down_grid_size = config.down_grid_size_y_ * config.down_grid_size_x_;
     if (grid_idx >= down_grid_size) {
       return;
     }
 
-    const auto yi = grid_idx / config.down_grid_size_x_;
-    const auto xi = grid_idx % config.down_grid_size_x_;
+    const auto yi = config.downsample_factor_ * (grid_idx / config.down_grid_size_x_);
+    const auto xi = config.downsample_factor_ * (grid_idx % config.down_grid_size_x_);
+    std::size_t array_idx = yi * config.grid_size_x_ + xi;
 
     int32_t label = -1;
     float max_score = -1;
     for (std::size_t ci = 0; ci < config.class_size_; ci++) {
-      float score = sigmoid(out_heatmap[down_grid_size * ci + grid_idx]);
+      float score = sigmoid(out_heatmap[grid_size * ci + array_idx]);
```

Comment:
`out_heatmap` is of size `down_grid_size_x_ * down_grid_size_y_ * class_size_`. I think this might end with out of boundary access. 

```
 const auto grid_xy_size = config_.down_grid_size_x_ * config_.down_grid_size_y_;
.....
  head_out_heatmap_d_ = cuda::make_unique<float[]>(grid_xy_size * config_.class_size_);
```